### PR TITLE
Dagger 2 framework is added for dependancy injection

### PR DESCRIPTION
**Pull request by jalalkhademi**
I just added dagger 2 to app/build.gradle file. So it is possible to use @Inject and @Component annotations for injecting dependencies

### CODE REVIEW

**Comment by joanb** [comment]
Hey! thanks for your contribution :)
Anyway, following good practices, it's not recommended to add libraries without making use of them. It would be nice if you could inject the presenter in the Activity. 
If you have doubt's, don't hesitate to ask! 
Also, if you want to experiment with something new, you could give a try to Koin. Both Libraries are welcome in this project.
Happy coding 😄

**Comment by jalalkhademi** [comment]
Cool I didnt know to inject which dependency into which class. Btw it is done now. I injected presenter into ClocksActivity.

File: app/src/main/java/com/theopensourcefamily/chessclock/ClocksActivity.kt [hunk 1]
```diff
@@ -3,12 +3,20 @@ package com.theopensourcefamily.chessclock
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
 import com.jakewharton.rxbinding3.view.clicks
+import dagger.Component
 import io.reactivex.Observable
 import kotlinx.android.synthetic.main.activity_clocks.*
 
+@Component
+interface ClockPresenterFactory{
```
Comment (joanb) [inline app/src/main/java/com/theopensourcefamily/chessclock/ClocksActivity.kt]: The linter here complains about the missing space:
```suggestion
interface ClockPresenterFactory {
```
A better place for this interface would be it's own file, outside the activity file.

File: app/src/main/java/com/theopensourcefamily/chessclock/ClocksActivity.kt [hunk 2]
```diff
@@ -3,12 +3,20 @@ package com.theopensourcefamily.chessclock
...
+  fun clockPresenter(): ClocksPresenter
+}
+
 class ClocksActivity : AppCompatActivity(), ClocksView {
 
-  val presenter = ClocksPresenter() // maybe DI here, with koin or wathever
+  // initialized by dagger 2. DaggerClockPresenterFactory class will be generated by
```
Comment (joanb) [inline app/src/main/java/com/theopensourcefamily/chessclock/ClocksActivity.kt]: You can remove this comments 😄

**Comment by joanb** [comment]
@all-contributorsbot please add @jalalkhademi for code

**Pull request merged**